streams/src/main/java/org/apache/kafka/streams/processor/internals/PunctuationQueue.java

```diff
@@ -16,16 +16,20 @@
  */
 package org.apache.kafka.streams.processor.internals;
 
+import org.apache.kafka.streams.processor.Cancellable;
+import org.apache.kafka.streams.processor.PunctuationType;
+
 import java.util.PriorityQueue;
 
 public class PunctuationQueue {
 
     private final PriorityQueue<PunctuationSchedule> pq = new PriorityQueue<>();
 
-    public void schedule(PunctuationSchedule sched) {
+    public Cancellable schedule(PunctuationSchedule sched) {
```

Comment:
I did it this way so that that StreamTask#schedule method can return from each case statement removing the need to use "break" each time, which is error prone IMO. Of course given how thorough code review are over here it might have been over-defensive of me ;-) It's also more conducive to functional style in the caller. I imagine a time when java supports pattern matching and that switch statement can be turned into a nice expression. But that is science-fiction for now. Do you want me to change it or leave as is?